src/RouteCollectorFactory.php

```diff
@@ -55,7 +55,7 @@ private function getDetectDuplicatesFlag(ContainerInterface $container): bool
             return true;
         }
 
-        $config = $container->get('config');
+        $config = (array)$container->get('config');
```

Comment:
probably better with instanceof `ArrayObject` check, eg:

```suggestion
        $config = $container->get('config');
        $config = $config instanceof \ArrayObject
            ? $config->getArrayCopy()
            : $config;
```

Reply:
If we support a common interface, the implementation may be different. 
so, maybe other implementations will implement custom collections, like: MyArrayWrapper or ArrayIterator and etc.

If we use `$config instanceof \ArrayObject` we will close only 2 case: with simple array and ArrayObject.

Reply:
Array casts on objects are not usage of object public API, and expose object internals in unintentional ways. Operating with the `ArrayAccess` API (and checking for that type) is more correct here.

Reply:
@Ocramius, I fixed its like @samsonasik proposed, however with ArrayAccess.